<launch>
    <arg name="ns" default="motor_hw_interface" />
    <arg name="port" default="/dev/mcu" />
    <arg name="baud" default="921600" />
    <arg name="tf_prefix" default="" />
    <arg name="loop_hz" default="500" />
    <arg name="controller_config" default="$(find damiao_motor_driver)/config/controllers.yaml" />

    <group ns="$(arg ns)">
        <node pkg="damiao_motor_driver"
              type="motor_hw_interface_node"
              name="motor_hw_interface"
              output="screen">
            <param name="port" value="$(arg port)" />
            <param name="baud" value="$(arg baud)" />
            <param name="tf_prefix" value="$(arg tf_prefix)" />
            <param name="loop_hz" value="$(arg loop_hz)" />

            <rosparam command="load" file="$(arg controller_config)" />
        </node>

        <node pkg="controller_manager"
              type="spawner"
              name="controller_spawner"
              output="screen"
              args="joint_state_controller effort_group_controller" />
    </group>
</launch>
